examples/greeter-extension/README.md

```diff
@@ -0,0 +1,304 @@
+     * 'greeter')
+     */
+    @inject.getter(bindingTagFilter({extensionPoint: 'greeter'}))
+    private greeters: Getter<Greeter[]>,
+  ) {}
+  // ...
+}
+```
+
+To customize metadata such as `id` for the extension point, we can use
+`@extensionPoint` to decorate the class, such as:
+
+```ts
+@extensionPoint('greeter')
+export class GreeterExtensionPoint {}
+```
+
+#### Access extensions for a given extension point
+
+To simplify access to extensions for a given extension point, we use dependency
+injection to receive a `getter` function that gives us a list of greeters.
+
+```ts
+@extensionPoint('greeter')
+export class GreeterExtensionPoint {
+  constructor(
+    /**
+     * Inject a getter function to fetch greeters (bindings tagged with
+     * 'greeter')
+     */
+    @extensions() // Sugar for @inject.getter(filterByTag({extensionPoint: 'greeter'}))
+    private greeters: Getter<Greeter[]>, // ...
+  ) {}
+}
+```
+
+Please note that it's possible to add/remove greeters after the extension point
+is instantiated. The injected getter function always pick up the latest list of
+greeters. For example:
+
+```ts
+// Get the latest list of greeters
+const greeters = await this.greeters();
+```
+
+#### Implement the delegation logic
+
+Typically, the extension point implementation will get a list of registered
+extensions. For example, when a person needs to be greeted in a specific
+language, the code iterates through all greeters to find an instance that
+matches the language.
+
+```ts
+/**
+ * Greet in the given language
+ * @param language Language code
+ * @param name Name
+ */
+async greet(language: string, name: string): Promise<string> {
+  let greeting: string = '';
+  // Get the latest list of greeters
+  const greeters = await this.greeters();
+  // Find a greeter that can speak the given language
+  for (const greeter of greeters) {
+    if (greeter.language === language) {
+      greeting = greeter.greet(name);
+      break;
+    }
+  }
+  if (!greeting) {
+    // Fall back to English
+    greeting = `Hello, ${name}`;
+  }
+  if (this.options && this.options.color) {
+    greeting = chalk.keyword(this.options.color)(greeting);
+  }
+  return greeting;
+}
+```
+
+## Implement an extension
+
+Modules that want to connect to `greeter` extension point must implement
+`Greeter` interface in their extension. The key attribute is that the
+`GreeterExtensionPoint` being extended knows nothing about the module that is
+connecting to it beyond the scope of that contract. This allows `greeters` built
+by different individuals or companies to interact seamlessly, even without their
+knowing much about one another.
+
+```ts
+import {Greeter, asGreeter} from '../types';
+import {bind, inject} from '@loopback/context';
+
+/**
+ * Options for the Chinese greeter
+ */
+export interface ChineseGreeterOptions {
+  // Name first, default to `true`
+  nameFirst: boolean;
+}
+
+/**
+ * A greeter implementation for Chinese
+ */
+@bind(asGreeter)
```

Comment:
Any chance we can describe what `asGreeter` is earlier? It is only explained at line 223.

Reply:
![image](https://user-images.githubusercontent.com/6864736/54716161-89dab700-4b2b-11e9-8ef5-2ad0188e91d2.png)

If `@bind(asGreeter)` is : 

![image](https://user-images.githubusercontent.com/6864736/54716269-bee70980-4b2b-11e9-8dba-3925dcc3f0ad.png)

then what does

![image](https://user-images.githubusercontent.com/6864736/54716329-dde59b80-4b2b-11e9-813c-40fe62bb8e17.png)
(adjusted for ChineseGreeter)

do exactly?

Reply:
I made some adjustment.